src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1118,31 +1154,33 @@ public RequestLineStatus TakeStartLine(SocketInput input)
                     {
                         // No need to allocate an extra string if the path didn't need
                         // decoding and there's no query string following it.
-                        rawTarget = requestUrlPath;
+                        rawUrlPath = requestUrlPath;
                     }
                     else
                     {
-                        rawTarget = pathBegin.GetAsciiString(ref queryEnd);
+                        rawUrlPath = pathBegin.GetAsciiString(ref queryEnd);
                     }
                 }
 
-                var normalizedTarget = PathNormalizer.RemoveDotSegments(requestUrlPath);
+                var normalizedUrlPath = requestUrlPath == null
+                    ? string.Empty
+                    : PathNormalizer.RemoveDotSegments(requestUrlPath);
 
                 consumed = scan;
                 Method = method;
                 QueryString = queryString;
-                RawTarget = rawTarget;
+                RawTarget = requestUriScheme + requestAuthority + rawUrlPath;
```

Comment:
I think you'll see a perf hit from these concatenations. Ideally you should get the raw target in one go and set it here, instead of building it from it's parts.